### PR TITLE
Ember tests

### PR DESCRIPTION
**Pull request by Robdel12**
# What is this?

I wrote some tests... This tests creating a blog post, creating a new about, and making a contact request.

*This also includes an Ember CLI update*

### CODE REVIEW

File: ember/app/controllers/posts/base.js [hunk 1]
```diff
@@ -27,8 +27,6 @@ export default Ember.Controller.extend({
 
       if(this.get("model.isPublished") === true){
         this.transitionToRoute("posts.show", this.get("model.postSlug"));
-      } else {
-        this.transitionToRoute("posts.edit", this.get("model.postSlug"));
```
Comment (Robdel12) [inline ember/app/controllers/posts/base.js]: I should address this.

- [ ] TODO: Either add this back or add auto save to the "new" route

File: ember/tests/acceptance/adding-about-test.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  };
+
+  visit('/about/new');
+  fillIn(".title", timeline.title);
+  fillIn(".date", timeline.eventDate);
+  fillIn(".description", timeline.description);
+  fillIn("#isPublished", timeline.isPublished);
+  click("button:contains('Save')");
+
+  pretender.post('api/timeline', function() {
```
Comment (Robdel12) [inline ember/tests/acceptance/adding-about-test.js]: - [ ] TODO:  Test response data

File: ember/tests/acceptance/adding-post-test.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    return [200, { 'Content-Type': 'application/json' }, JSON.stringify({
+      post: [post],
+      meta: { total_pages: 1 }
+    })];
+  });
+
+  andThen(function() {
+    ok(find('h3:contains("'+ post.title +'")').length, 'expected to see "My new post"');
+  });
+});
```
Comment (Robdel12) [inline ember/tests/acceptance/adding-post-test.js]: - [ ] TODO: There are more tests I could write for this

File: ember/tests/acceptance/adding-post-test.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  visit('/posts/new');
+  fillIn('#post_title', post.title);
+  fillIn('#excerpt', post.excerpt);
+  fillIn('.post-text-area', post.body);
+  fillIn('select', post.isPublished);
+  click('button:contains("Edit post")');
+
+  pretender.post('api/posts', function(req) {
+    return [201, { 'Content-Type': 'application/json' }, JSON.stringify({
+      posts: post
```
Comment (Robdel12) [inline ember/tests/acceptance/adding-post-test.js]: - [ ] TODO: Test response

File: ember/tests/acceptance/adding-post-test.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    return [200, { 'Content-Type': 'application/json' }, JSON.stringify({
+      post: [post],
+      meta: { total_pages: 1 }
+    })];
+  });
+
+  andThen(function() {
+    ok(find('h3:contains("'+ post.title +'")').length, 'expected to see "My new post"');
+    ok(find('span:contains("'+ post.excerpt +'")').length, 'expected to see "This is my excerpt"');
+    ok(find(!!$(".inner-date").text(), 'expected to see "This is my excerpt"'));
```
Comment (Robdel12) [inline ember/tests/acceptance/adding-post-test.js]: - [ ] TODO: Fix error message

File: ember/app/controllers/contact.js [hunk 6]
```diff
@@ -49,13 +49,13 @@ export default Ember.ObjectController.extend({
       });
 
       this.set("modalClosed", false);
-      Ember.$("body").addClass("modal-backing");
+      Ember.$(".main-container").addClass("modal-backing");
```
Comment (sergiocruz) [inline ember/app/controllers/contact.js]: Is this how you leverage jQuery within Ember? Sorry just trying to learn Ember from your code :)

Comment (Robdel12) [inline ember/app/controllers/contact.js]: Yeah, this is actually a _bad practice_ but I'm not quite sure how to toggle a class on the body without it :scream:

File: ember/tests/acceptance/adding-about-test.js [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    })];
+  });
+
+  pretender.get('api/timeline', function() {
+    return [201, { 'Content-Type': 'application/json' }, JSON.stringify({
+      timelines: [timeline]
+    })];
+  });
+
+  andThen(function() {
```
Comment (tracehelms) [inline ember/tests/acceptance/adding-about-test.js]: https://www.youtube.com/watch?v=Kp5nUQkgWIc

On a serious note, Ember's testing suite is actually pretty cool.

File: ember/tests/acceptance/adding-about-test.js [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+    eventDate: new Date(2014, 02, 03), //March 3rd, 2014
+    description: "My new abouts description",
+    isPublished: true
+  };
+
+  visit('/about/new');
+  fillIn(".title", timeline.title);
+  fillIn(".date", timeline.eventDate);
+  fillIn(".description", timeline.description);
+  fillIn("#isPublished", timeline.isPublished);
```
Comment (Robdel12) [inline ember/tests/acceptance/adding-about-test.js]: Does this work?

File: ember/tests/acceptance/adding-post-test.js [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+    post_slug: "my-new-post",
+    is_published: true,
+    published_date: new Date(2015, 02, 03)
+  };
+
+  visit('/posts/new');
+  fillIn('#post_title', post.title);
+  fillIn('#excerpt', post.excerpt);
+  fillIn('.post-text-area', post.body);
+  // fillIn('.published-date', post.published_date);
```
Comment (Robdel12) [inline ember/tests/acceptance/adding-post-test.js]: - [ ] TODO: Fix this. Figure out why it no work.

**Pull request merged**